pkg/descheduler/descheduler.go

```diff
@@ -75,7 +76,21 @@ func Run(ctx context.Context, rs *options.DeschedulerServer) error {
 		<-ctx.Done()
 		close(stopChannel)
 	}()
-	return RunDeschedulerStrategies(ctx, rs, deschedulerPolicy, evictionPolicyGroupVersion, stopChannel)
+
+	runFn := func() error {
+		return RunDeschedulerStrategies(ctx, rs, deschedulerPolicy, evictionPolicyGroupVersion, stopChannel)
+	}
```

Comment:
It feels a bit odd to be wrapping the function like this, is there a different way we could pass it to `leader.NewLeaderElection` with all of the parameters intact?

Reply:
Had same feeling. But couldn't find a better way to pass `RunDeschedulerStrategies()` func to `leader.NewLeaderElection`, any ideas and advices here?

Reply:
Does `NewLeaderElection` need to be under a new directory? The function might go under `pkg/descheduler/descheduler.go` or a new `pkg/descheduler/leaderelection.go`. I do not see the `NewLeaderElection` function to be used anywhere but here so it could be just inlined:
```
if rs.LeaderElection.LeaderElect {
		if rs.DeschedulingInterval.Seconds() > 0 {
			...
			leaderelection.RunOrDie(ctx, leaderelection.LeaderElectionConfig{
					Lock:            lock,
					ReleaseOnCancel: true,
					LeaseDuration:   LeaderElectionConfig.LeaseDuration.Duration,
					RenewDeadline:   LeaderElectionConfig.RenewDeadline.Duration,
					RetryPeriod:     LeaderElectionConfig.RetryPeriod.Duration,
					Callbacks: leaderelection.LeaderCallbacks{
						OnStartedLeading: func(ctx context.Context) {
							klog.V(1).InfoS("Started leading")
							err := RunDeschedulerStrategies(ctx, rs, deschedulerPolicy, evictionPolicyGroupVersion, stopChannel)
							if err != nil {
								klog.Error(err)
							}
						},
			...
			return nil
		}
		...
}
```

Reply:
We removed the `leader` package and move it into `leaderelection.go` file. I think `NewLeaderElection` function makes things cleaner in a separated file, and it would a bit verbose to put entire logic in the `descheduler.go` file. But I agree with you, we don't use _that_ function again anywhere. 🤔

Reply:
Instead of nesting the `interval` check inside the `leaderElect` check, you could break both of those into an error check above. That might open up an option to make this flow better. Something like:
```go
if rs.LeaderElection.LeaderElect && rs.DeschedulingInterval.Seconds() == 0 {
  return fmt.Errorf("LeaderElection must be used with deschedulingInterval...")
}
if rs.LeaderElection.LeaderElect {
  // start leader election call
} else {
  // start normal run call
}
```

something like that. it's a little redundant but it makes the code clearer and removes the deep nesting